### DISCUSSION TITLE
Custom mapping for WooCommerce?

### DISCUSSION THREAD

**Discussion by robertsheie** [Sellbrite/Sellbrite-API#12] [Q&A] [answered]
Is it possible to create custom mappings from attributes in Sellbrite to attributes in a WooCommerce store?

**Comment by rengler-godaddy** [comment]
Hello - Thanks for reaching out to us! 

Sellbrite currently doesn't support custom attributes from WooCommerce. You would have to add or update any custom attributes directly on WooCommerce.

**Reply by robertsheie** [reply]
Is there a custom API solution to this problem?

I am aware that Sellbrite doesn't support WooCommerce custom attributes. Years ago, an SB support person told me it would have to be a custom integration. That's what I'm asking about here.

**Reply by Taegost** [reply]
I'm working on getting custom attribute sync'ing to work myself, I actually have #11 open to address a problem I found with the documentation.
If you're trying to pull custom attributes out of Sellbrite to update in WooCommerce, that's relatively easy as the Sellbrite endpoint Product `GET` as mentioned by @jengler-godaddy will get the custom attributes. It can then be parsed to and sent to the WooCommerce API.
If you're trying to pull the attribute from WooCommerce to put into Sellbrite, that one isn't so easy as the Sellbrite API documentation for the `PUT` is wrong and it hasn't been updated yet, so it's unknown how to actually do that in their API.

**Reply by Taegost** [reply]
As for the "custom integration" portion of your question, there are some automation tools out there like n8n or Zapier that can facilitate pushing and pulling the data from/to the API endpoints without having to go too deep into the weeds of programming.

**Reply by robertsheie** [reply]
What does this look like from a workflow perspective as a user publishing new products? If it's practically useful, then how would I find a developer to implement this? I am still interested in making this work, but have no API or development experience.

**Reply by Taegost** [reply]
Well, now that I've gotten the Sellbrite side of this figured out (we were able to work through it in my issue #11), I can say it's possible to push/pull the attributes between Sellbrite and WooCommerce and it's not that difficult, but it's also not very intuitive (it doesn't help that the documentation update that came out of the resolution to my inquiry is almost worse than the wrong information that was in it before).

The simplified workflow would be: 
1) Make API call(s) to get data from Sellbrite 
2) Extract the needed information 
3) Make API call(s) to push data to WooCommerce

Steps 1 and 3 are relatively simple, step 2 can be tricky. If you have products with variations, there are 2 different places you have to look for the custom attributes (First is the root product itself, the second is each variation. They can all have different attributes).

It sounds complicated, but once you figure it out it makes a strange kind of sense.
As for getting it wired up, you may be able to get it working with [Zapier](https://zapier.com) without needing any coding experience. I personally use [n8n](n8n.io) because I can set it up for free on my home server. n8n requires a little bit of programming knowledge to make it work though, because the logic needed for step 2 has to be coded in either Javascript or Python.
I'm not sure how to go about looking for a developer to do the work (I've got too much on my plate to commit to something that important), but I'm more than happy to help provide guidance and answer questions.

**Reply by robertsheie** [reply]
Thank you for responding. Fortunately my products typically have no variations. I'm looking around in Zapier for the first time. Creating my first "zap" and wondering what trigger would handle your step 1 (make API call to get SB data). Webhooks? Catch Hook? 

Regarding developer help: what specific skills should I look for in hiring a freelance developer to do this?

**Reply by Taegost** [reply]
If you're pulling data from Sellbrite and pushing it to Wordpress (or anywhere else), I think you'd need to use either a manual or scheduled trigger. Unfortunately, according to discussion #4 Sellbrite doesn't have any webhooks available to use. For my automation, I typically do it on a schedule based on what I need (every 5 minutes to look for new orders, once a day at midnight to update custom attributes, etc).
If you're pulling data from Wordpress, you can usually set up a webhook that runs whenever a certain thing happens (i.e. a custom attribute is added/updated, new product created, etc.). I'm not sure how to set that up though, I actually just jumped from Wordpress and Woocommerce to Shopify right about the time I was going to start doing things like that.

The skills you'd be looking for in a developer would depend on how involved you want them to get, but at a minimum you'd want someone with experience using the automation tool (Zapier, if you choose to continue with it) and experience working with APIs. It doesn't necessarily need to be a lot of experience either, the hardest part about working with APIs like this tends to be 1) poor/outdated documentation of the API itself, 2) creating scripts to do the things that the tool can't, and 3) Setting up the authentication in the individual APIs can be a pain in the ass (I'm looking at you, Google Cloud!). Neither of those things require much skill or experience to overcome, it may just take someone with less experience a longer time to do the work.

Honestly, what I'd look for more in a developer for this is someone that is trustworthy and reliable, because you'll basically be giving them the literal keys to the kingdom.

**Comment by rengler-godaddy** [comment]
You can use the GET method mentioned [here](https://developer.sellbrite.com/endpoints/products) to export the custom attributes using the API, but there isn’t a POST or PUT method for publishing to WooCommerce using the Sellbrite API. You would need to use the WooCommerce API to push those attributes over. I did find some information in this WordPress post [here](https://wordpress.org/support/topic/create-a-product-with-attributes-using-rest-api/) that gives some helpful information on how that would work.

**Comment by robertsheie** [comment, marked as answer]
After quite a bit of trial and error, I was able to get this working without Zapier or a cron job.

The solution was a small WordPress plugin that runs after Sellbrite publishes a product to WooCommerce. The plugin:

- Uses the WooCommerce product SKU to retrieve the Sellbrite product via the Sellbrite API.
- Reads the Sellbrite fields (`brand`, `condition`) and `custom_attributes` (color, material, pattern, size, etc.).
- Creates missing WooCommerce attribute terms when necessary.
- Assigns the terms to the product and updates the product's `_product_attributes` metadata so the attributes appear correctly in WooCommerce.
- Skips empty values (for example, products without GTIN or MPN).

One important lesson: hooking into the correct WooCommerce REST product event was essential. My initial attempts using standard WordPress product save hooks were unreliable because Sellbrite publishes products through the WooCommerce REST API.

The end result is that all attribute data is entered only once in Sellbrite and is automatically available in WooCommerce after publishing. This eliminated the need to manually enter attributes in both systems.
